api.py

```diff
@@ -367,7 +389,133 @@ def set_generic_data(dataset_id, key, data_string, **kwargs):
     return ingest_entities(dataset_id, {'entities': [entity]}, **kwargs)
 
 
+def convert_coordinates(point):
+    if len(point) > 2:
+        raise KeyError('A point should only have two dimensions', point)
+    if 'lat' and 'lon' in point:
+        coord = {
+            'x': point['lon'],
+            'y': point['lat']
+        }
+    elif 'x' and 'y' in sample['point']:
+        coord = {
+            'x': point['x'],
+            'y': point['y']
+        }
+    else:
+        raise KeyError('invalid coordinate', point)
+
+    coord['z'] = 0
+
+    return coord
+
+
+def convert_samples_to_entity_set(sample_list):
+    conduce_keys = ['id', 'kind', 'time', 'point', 'path', 'polygon']
+    entities = []
+    for sample in sample_list:
+        if not isinstance(datetime, sample['time']):
+            raise TypeError('time must be a datetime object')
+        path = []
+        if 'point' in sample:
+            path = [convert_coordinates(sample['point'])]
+        if 'path' in sample:
+            if len(path) > 0:
+                raise KeyError('A sample may only contain one of point, path or polygon', sample)
+            if len(sample['path']) < 2:
+                raise KeyError('paths must have at least two points')
+            for point in sample['path']:
+                path.append(convert_coordinates(point))
+        if 'polygon' in sample:
+            if len(path) > 0:
+                raise KeyError('A sample may only contain one of point, path or polygon', sample)
+            if len(sample['path']) < 3:
+                raise KeyError('polygons must have at least three points')
+            for point in sample['polygon']:
+                path.append(convert_coordinates(point))
+
+        attribute_keys = [key for key in sample.keys() if key not in conduce_keys]
+        attributes = util.get_attributes(attribute_keys, sample)
```

Comment:
I reused `util.get_attributes` but it does conversions from float to int.  Take a look and see what you think.  I don't think it would be a problem to remove the conversion to int and leave all attributes in floating point.